### DISCUSSION TITLE
How to close watch?

### DISCUSSION THREAD

**Discussion by xdRight** [etcd-io/etcd#14496] [Q&A]
as shown in https://github.com/etcd-io/etcd/blob/v3.5.4/client/v3/watch.go
![image](https://user-images.githubusercontent.com/34833891/191218919-7b8c1eb9-66e8-4cbe-97c2-4d5a425d4514.png)

Cloud you tell me a another method to close watch? Thanks!

**Comment by SimFG** [comment]
you can try `close(context.Done())`, the stream will be close.
![image](https://user-images.githubusercontent.com/21985684/191226887-037b96b6-57e8-4020-94f0-6ed946d9805e.png)

**Comment by xdRight** [comment]
I new a watcher. like 

`watcher := clientv3.Watcher(client)
ctx, cancelFunc := context.WithCancel(context.TODO())
defer cancelFunc()
watchChan := watcher.Watch(ctx,"prifix")`

It seems to work.

**Reply by SimFG** [reply]
yes, it's same. The `cancel` function will close the `done` channel.
